src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ticketRuleForm.tsx

```diff
@@ -107,10 +112,20 @@ class TicketRuleForm extends React.Component<Props, State> {
         const options = choices.map(([value, label]) => ({value, label}));
         return resolve({options});
       }
+
       return this.debouncedOptionLoad(field, input, (err, result) => {
         if (err) {
           reject(err);
         } else {
+          const dynamicFieldChoices = result.options.map(obj => [obj.value, obj.label]);
+          this.setState(state => {
```

Comment:
@NisanthanNanthakumar is there a better way to do this?

Reply:
@mgaeta the spread operator on objects returns a shallow copy, which is not ideal for  multidimensional objects. We can use the `cloneDeep` and `set` functions from `lodash` to make a deep clone, update the cloned object and then setState with that object.

```suggestion
import cloneDeep from 'lodash/cloneDeep';
import set from 'lodash/set';

this.setState(prevState => {
  const newState = cloneDeep(prevState);
  set(newState, `dynamicFieldChoices[${field.name}]`, value);
  return newState;
});
```

Reply:
👍🏽